pkg/icons/icons.go

```diff
@@ -0,0 +1,149 @@
+package icons
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"helm.sh/helm/v3/pkg/repo"
+)
+
+const (
+	partnerFilePath     = "index.yaml"
+	partnerDownloadPath = "assets/logos"
+)
+
+// PackageIconOverride is a struct to hold the package name, path, and icon to override
+type PackageIconOverride struct {
+	Name string
+	Path string
+	Icon string
+}
+
+// PackageIconList is a list of PackageIconOverride
+type PackageIconList []PackageIconOverride
+
+// PackageIconMap is a map of PackageIconOverride
+type PackageIconMap map[string]PackageIconOverride
+
+// ParsePackageToPackageIconOverride will convert a Package from a PackageList to a PackageIcon for a PackageIconList
+func ParsePackageToPackageIconOverride(name, path, icon string) PackageIconOverride {
+	return PackageIconOverride{
+		Name: name,
+		Path: path,
+		Icon: icon,
+	}
+}
+
+// CheckFilesStructure will check if index.yaml and assets/logos exist
+// will log and exit if the file structure does not exist
+func CheckFilesStructure() {
+	if _, err := os.Stat(partnerFilePath); os.IsNotExist(err) {
+		fmt.Printf("File not found: %s\n", partnerFilePath)
+		logrus.Fatalf("File not found: %s\n", partnerFilePath)
+	}
+	_, err := os.Stat(partnerDownloadPath)
+	if os.IsNotExist(err) {
+		logrus.Fatalf("Directory not found: %s\n", partnerDownloadPath)
+	}
+}
+
+// CheckForDownloadedIcon will check if the package has an upstream.yaml file and if the iconURL is a local file
+func CheckForDownloadedIcon(packageName string) string {
```

Comment:
A couple thoughts on this function:
1. The comment is out of date, I think - it refers to `upstream.yaml`.
2. It would be better to use `filepath.Join()` rather than `fmt.Sprintf()` when setting `filePath`.
3. Consider using `Exists()` rather than `os.Stat()` for checking existence of `filePath`.
4. Nit: consider making `extensions` a constant at package scope to avoid allocating it on every call to `CheckForDownloadedIcons`. The name could be more specific as well, maybe `validIconExtensions` or something like that. Feel free to disregard this one since it doesn't really matter.

Reply:
I implemented 1,3 and 4 but 2 I don't see the point.